src/solvers/refinement/string_expr.h

```diff
@@ -0,0 +1,91 @@
+/******************************************************************\
+
+Module: String expressions for the string solver
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVERS_REFINEMENT_STRING_EXPR_H
+#define CPROVER_SOLVERS_REFINEMENT_STRING_EXPR_H
+
+#include <langapi/language_ui.h>
+
+#include <solvers/refinement/bv_refinement.h>
+#include <solvers/refinement/refined_string_type.h>
+
+
+// Expressions that encode strings
+class string_exprt: public struct_exprt
```

Comment:
The solver requires to have both an array and its length to encode a string, so I thought a structure with these two fields was the most natural. We don't really exploit this in the solver and there just inheriting from `exprt` would be okay, but when we will introduce them in the goto program it ensures that they are processed correctly by the other components of CBMC (ie like struct expressions).